core/server/internal/response/writer.go

```diff
@@ -10,13 +10,18 @@ import (
 // Writer is a custom wrapper around http.ResponseWriter used in the server package.
 // It also captures the HTTP status code and and implements the http.Flusher and http.Hijacker interfaces.
 type Writer struct {
+	wroteHeader bool
 	http.ResponseWriter
 	Status int
 }
 
 // WriteHeader sets the status code and calls the WriteHeader() method of http.ResponseWriter.
 func (w *Writer) WriteHeader(statusCode int) {
 	w.Status = statusCode
+	if !w.wroteHeader {
```

Comment:
@tatang26 did you test this behavior? Looks reverse to me, WDYT?

Reply:
Oh I'm sorry this change must not be here, I was looking a way to avoid the `http: superfluous response.WriteHeader call` warning, that need to be covered by tests, I'll remove it from this PR

Reply:
@tatang26 It looks like it's reversed IMO.

Reply:
Yup, you're right that was a mistake, the negation must be removed